metadata-store-directory/src/main/java/org/apache/helix/metadatastore/server/MetadataStoreDirectoryServer.java

```diff
@@ -0,0 +1,232 @@
+package org.apache.helix.metadatastore.server;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.net.ssl.SSLContext;
+
+import org.apache.helix.metadatastore.server.common.ContextPropertyKeys;
+import org.apache.helix.metadatastore.server.common.RestNamespace;
+import org.apache.helix.metadatastore.server.common.ServletType;
+import org.apache.helix.metadatastore.server.auditlog.AuditLogger;
+import org.apache.helix.metadatastore.server.filters.AuditLogFilter;
+import org.apache.helix.metadatastore.server.filters.CORSFilter;
+import org.eclipse.jetty.http.HttpVersion;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.HttpConnectionFactory;
+import org.eclipse.jetty.server.SecureRequestCustomizer;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.server.SslConnectionFactory;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.server.ServerProperties;
+import org.glassfish.jersey.servlet.ServletContainer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class MetadataStoreDirectoryServer {
```

Comment:
This is almost same as REST module. Can we think about how to reduce the redundancy of the code?

Reply:
This is because this is a REST deployable itself. 

We could first make it work/run as a standalone, and think about cutting down more things as necessary. I've tried to remove as much unnecessary code as possible.

If you have concrete suggestions, please post them here.